src/lib/mongoBackend/safeMongo.cpp

```diff
@@ -243,12 +244,12 @@ bool nextSafeOrError(const std::auto_ptr<DBClientCursor>& cursor, BSONObj* r, st
   }
   catch (const std::exception &e)
   {
-    *err = e.what();
+    *err = std::string(e.what()) + " at " + caller;
     return false;
   }
   catch (...)
   {
-    *err = "generic exception";
+    *err = "generic exception at " + caller;
```

Comment:
I still can't understand how this works ...

Perhoaps like this:

1) *err = "generic exception at "  => result is a std::string
2) append 'caller' to the string